xc-tool-ts/src/config.ts

```diff
@@ -0,0 +1,157 @@
+import { PythUpgradable, PythUpgradable__factory } from "./evm/bindings/index";
+
+import * as ethers from "ethers";
+import * as solWeb3 from "@solana/web3.js";
+import * as borsh from "borsh";
+
+export interface Config {
+  mainnet: MainnetConfig;
+  testnet: TestnetConfig;
+  localDevnet: LocalDevnetConfig;
+}
+
+export interface NetworkConfig {
+  getEvmConfigs(): EvmConfig[];
+  getSolConfigs(): SolConfig[];
+}
+
+export class MainnetConfig implements NetworkConfig {
+  getEvmConfigs(): EvmConfig[] {
+    return [];
+  }
+
+  getSolConfigs(): SolConfig[] {
+    return [];
+  }
+}
+
+export class LocalDevnetConfig implements NetworkConfig {
```

Comment:
having separate config classes for mainnet / testnet / etc. seems wrong to me. I think a better solution would be a single config class that was flexible enough to be used across all of them. I think a class with fields like`evmConfigs: Map<string, EvmConfig>` would be sufficient? (the key of the map would be the human-readable name of the network.)

Reply:
I agree, if you make it more generalized then we can move the actual config to a config file. Inside config files you can also have a `cluster` field to specify what cluster this is. 

Then, the config file could look like this (just an idea):

```yaml
target_chains:
  - name: bnb_testnet
    chain_type: evm
    cluster: testnet
    rpc: https://....
    contract: 0xabde000
 - ....
```

Reply:
I think that using a sacred config file with all the values is a very bad idea. IMO a better approach is to keep up-to-date defaults and only splice in temporary changes, such as "defaults but RPC for Pythnet is different". Such partial snippets would be treated as exceptions instead of the norm. The use of an altered config can then be detected and flagged with the user when the tool is run, instead of guesswork comparing people's configs.

I also think the clear divide between networks is very important.  For example, there's different things on Pythnet than on Solana. SOL has no remote executor, Pythnet has no governance contract. Instead of using a nullable that can mistakenly be filled or left empty, I find it far more helpful to make an invalid choice unrepresentable, by never mentioning remote executor on Solana, and vice versa with governance contract on Pythnet.